### PR TITLE
add new rule for azure disk management

### PR DESCRIPTION
**Pull request by shahinshirvani**

### CODE REVIEW

File: azure/iac/disks.rego [hunk 1]
```diff
@@ -6,9 +6,9 @@ package rule
 # PR-AZR-ARM-DSK-001
 #
 
-default disk_encrypt = null
+default disk_encryption = null
```
Comment (rezoan) [inline azure/iac/disks.rego:9]: @shahinshirvani if you are renaming the rule, make sure to rename variable parameter as well with the same name. example at line 19.

File: azure/iac/disks.rego [hunk 2]
```diff
@@ -79,3 +79,102 @@ disk_encrypt_metadata := {
...
+    count(resource.properties.encryption.diskEncryptionSetId) == 0
+    metadata:= {
+        "resource_path": [["resources",i,"properties","encryption","diskEncryptionSetId"]]
+    }
+}
+
+azure_issue["disk_encryption_2"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.compute/disks"
+    lowe(resource.properties.encryption.type) != "encryptionatrestwithcustomerkey"
```
Comment (rezoan) [inline azure/iac/disks.rego:137]: @shahinshirvani this line should be `lowe(resource.properties.encryption.type) == "encryptionatrestwithplatformkey"`
Ref: https://docs.microsoft.com/en-us/azure/templates/microsoft.compute/disks?tabs=bicep#encryption

**Review by rezoan — CHANGES_REQUESTED** [review]

**Comment by rezoan** [comment]
@shahinshirvani also please make sure to target the master branch. we are no longer using the development branch for this repo.

File: azure/iac/disks.rego (hunk 1 above)
Comment (shahinshirvani) [reply azure/iac/disks.rego:9]: fixed

File: azure/iac/disks.rego (hunk 2 above)
Comment (shahinshirvani) [reply azure/iac/disks.rego:137]: @rezoan according to this: https://docs.microsoft.com/en-us/azure/virtual-machines/disks-enable-customer-managed-keys-portal#set-up-your-disk-encryption-set
it should be lower(resource.properties.encryption.type) == "encryptionatrestwithcustomerkey"

Comment (rezoan) [reply azure/iac/disks.rego]: @shahinshirvani `EncryptionAtRestWithCustomerKey `and `EncryptionAtRestWithPlatformAndCustomerKeys `both are considerate as encryption with custom key. so in your case there will be an azure issue if only the encryption type is `EncryptionAtRestWithPlatformKey`
See the available types at https://docs.microsoft.com/en-us/azure/templates/microsoft.compute/disks?tabs=bicep#encryption

The document you shared does not tell any different.
image
Goto the azure portal and you will see all the options there.

File: azure/iac/disks.rego [hunk 3]
```diff
@@ -79,3 +79,102 @@ disk_encrypt_metadata := {
...
+    count(resource.properties.encryption.diskEncryptionSetId) == 0
+    metadata:= {
+        "resource_path": [["resources",i,"properties","encryption","diskEncryptionSetId"]]
+    }
+}
+
+azure_issue["disk_encryption_2"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.compute/disks"
+    lower(resource.properties.encryption.type) != "encryptionatrestwithcustomerkey"
```
Comment (rezoan) [inline azure/iac/disks.rego:137]: @shahinshirvani EncryptionAtRestWithCustomerKey and EncryptionAtRestWithPlatformAndCustomerKeys both are considerate as encryption with custom key. so in your case there will be an azure issue if only the encryption type is EncryptionAtRestWithPlatformKey
See the available types at https://docs.microsoft.com/en-us/azure/templates/microsoft.compute/disks?tabs=bicep#encryption

The document you shared does not tell any different.
image
Goto the azure portal and you will see all the options there.

**Review by rezoan — CHANGES_REQUESTED** [review]

File: azure/iac/disks.rego (hunk 2 above)
Comment (shahinshirvani) [reply azure/iac/disks.rego]: I just got your point
fixed

**Review by rezoan — APPROVED** [review]

**Pull request merged**